runtime/dart_init.cc

```diff
@@ -100,6 +100,7 @@ static const char* kDartCheckedModeArgs[] = {
     "--enable_type_checks",
     "--error_on_bad_type",
     "--error_on_bad_override",
+    "--assert_initializer",
```

Comment:
Does this actually enable to asserts or just configure the parser to accept the syntax?  I would have expected it to go in kDartLanguageArgs.